test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -598,6 +598,17 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthFo
           expect(test.form.passwordToggleContainer().length).toBe(1);
         });
       });
+      itp('Triggers a showPassword event when show password button is clicked', function () {
+        return setup({ 'features.showPasswordToggleOnSignInPage': true }).then(function (test) {
+          var eventSpy = jasmine.createSpy('eventSpy');
+          test.router.on('showPassword', eventSpy);
+          test.form.setPassword('testpass');
+          test.form.setUsername('testuser');
+          expect(test.form.passwordToggleContainer().length).toBe(1);
+          test.form.passwordToggleShowContainer().click();
+          expect(eventSpy).toHaveBeenCalled();
```

Comment:
Since this event is intended to **only** fire when going from hidden -> shown state, we need to have a test that asserts the opposite case.

This makes the behavior explicitly clear (and will catch devs who might change [`click .button-show`](https://github.com/okta/okta-signin-widget/pull/486/files#diff-19b949c50e99d13eb55db7490ef2effdR101) to `click .button-hide` and expect the same result).